src/Cli/dotnet/Commands/Run/RunCommand.cs

```diff
@@ -463,6 +478,22 @@ private static void ThrowUnableToRunError(ProjectInstance project)
             throw new GracefulException(CliCommandStrings.RunCommandExceptionNoProjects, projectFileOrDirectoryPath, "--project");
         }
 
+        if (entryPointFilePath is ReadFromStdin)
```

Comment:
What about detecting whether there's input likes

```cs
Console.IsInputRedirected && Console.In.Peek() != -1
```

so that we could use likes

```cs
'Console.WriteLine("Hello World");' | dotnet run
```

Reply:
Oh, sorry, I get it know, you are suggesting `dotnet run` (with no argument, not even the hyphen `-`) should detect there is input. Well, that shouldn't read the code from stdin IMO, because the target program might want to read the stdin in that case.

Reply:
yes, I'm suggesting reading input from stdin when there's no csproj here since we would use the csproj file under the current folder by default, no sure if any other cases missed or would break

Reply:
I see, that would work I guess but I'm thinking it might be better to be more intentional anyway by requiring the hyphen `-`. We could relax that in the future as a follow up.